CSharpPatterns/CSharpPatterns/Decorator/ExtraLoudGigantosaurusDecorator.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CSharpPatterns.Decorator
+{
+    // Additional functionality that will be added to the main class and "wrap" it,
+    // thereby implementing the open-closed principle (Concrete Decorator)
+    public class ExtraLoudGigantosaurusDecorator : IGigantosaurus
+    {
+        private Gigantosaurus _gigantosaurus;
```

Comment:
The field should be declared with the component interface type to allow linking to concrete components as well as decorators. The same for constructor.